pdf/pdfcanvas.cpp

```diff
@@ -267,22 +278,48 @@ void PDFCanvas::layout()
     update();
 }
 
+static qreal dist2FromRect(const QRectF &rect, const QPoint &point)
+{
+    qreal dist = 0.;
+
+    if ((qreal)point.x() < rect.x()) {
+        dist += (rect.x() - (qreal)point.x()) * (rect.x() - (qreal)point.x());
+    } else if ((qreal)point.x() > rect.x() + rect.width()) {
```

Comment:
Maybe use left(), right(), top(), bottom() instead of calculating such values yourself?